hashes/src/sha256t.rs

```diff
@@ -6,20 +6,20 @@
 use core::marker::PhantomData;
 use core::ops::Index;
 use core::slice::SliceIndex;
-use core::{cmp, str};
+use core::str;
 
 use crate::{sha256, FromSliceError};
 
 type HashEngine = sha256::HashEngine;
 
 /// Trait representing a tag that can be used as a context for SHA256t hashes.
-pub trait Tag {
+pub trait Tag: Copy + core::hash::Hash + Ord {
     /// Returns a hash engine that is pre-tagged and is ready to be used for the data.
     fn engine() -> sha256::HashEngine;
 }
 
 /// Output of the SHA256t hash function.
-#[repr(transparent)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
In 923582c6d549e98f03559a5d359fccb9179d9cb9:

Why drop the `repr(transparent)`?

Reply:
Ah woops, that was just while I was playing around, didn't mean to commit that.